docs/troubleshooting.md

```diff
@@ -0,0 +1,69 @@
+
+## I have already installed `azureserviceoperator` but I'm not sure which service principal secret is in use?
+This command might be able to help you see your secret locally. 
+```kubectl get secret azureoperatorsettings -n operators -o go-template='{{range $k,$v := .data}}{{printf \"%s: \" $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{\"\n\"}}{{end}}'```
+[SO thread for reference](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret/58117444#58117444)
+
+## What if I see the operator namespace already exists for the cluster?
+
+If you see that the operator namespace already exists for the cluster, it's likely the result of trying to install the operator previously. It also means that you can now successfully move on to the next step. 
+However, this can also be a symptom of incorrectly installing the operator, for example, trying to run the install guide out of order. Other common issues related to an "out-of-order install":
+
+* `namespace already exist` - This means that azure-service-operator (ASO) has already been run for the cluster before.
+* `service-principal` issues: If you're seeing silent failures with service principal in use, meaning it is expired or needs to be reset, you can follow the steps as mentioned [here](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principal) or in step 4 here in the [OperatorHub guidance](https://operatorhub.io/operator/azure-service-operator).
+* `kubectl api-resource` failure with `error: unable to retrieve the complete list of server APIs: metcs.k8s.io/vabeta=1: the server is currently unable to handle the request.` : This error indicates a networking error within your cluster and is not related to your Azure Service Operator installation or configuration.
+
+ ## When trying to install cert-manager, I'm seeing that the connection is refused for `cert-manager-package`?
+This could be related to following open error: https://github.com/jetstack/cert-manager/issues/2752
+ 
+ ## The azureoperator-controller-manager seems to be in a crash loop. How do I investigate?
+
+`kubectl describe  pod operators`
+
+To get logs in case ASO operator is crashlooping : `kubectl logs -n  operators pod/azureoperator-controller-manager-****** --container manager`
+
+## What if I see cert-manager-webhook timeout errors?
+
+This is network related, and you should check to see if there are any ongoing network issues. This is evidenced by common error messages like: `Post https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=30s: connection refused` 
+## What if I am seeing APIserver errors like `metrics.k8s.io/v1beta1: the server is currently unable to handle request`?
+
+This could be a result of a Kubernetes internal error like control plane & data plane not being able to communicate. Follow normal Kubernetes troubleshooting steps.
```

Comment:
```suggestion
This could be a result of a Kubernetes internal error like the control plane & data plane not being able to communicate. Follow normal Kubernetes troubleshooting steps.
```